packages/blaze/lookup.js

```diff
@@ -36,40 +36,30 @@ var bindDataContext = function (x) {
 Blaze._OLDSTYLE_HELPER = {};
 
 Blaze._getTemplateHelper = function (template, name, tmplInstanceFunc) {
-  // XXX COMPAT WITH 0.9.3
-  var isKnownOldStyleHelper = false;
-
   if (template.__helpers.has(name)) {
-    var helper = template.__helpers.get(name);
+    let helper = template.__helpers.get(name);
     if (helper === Blaze._OLDSTYLE_HELPER) {
-      isKnownOldStyleHelper = true;
-    } else if (helper != null) {
-      return wrapHelper(bindDataContext(helper), tmplInstanceFunc);
+      throw new Meteor.Error("not-suported", "We removed support for old style templates");
```

Comment:
Pardon me, but what does improving exceptions or allowing setting a custom handler function has to do with disabling old template helpers?

Reply:
if you read a bit further in the code it will basically raise a deprecated message. @StorytellerCZ should I restore this? We can drop old style helpers in 3.0 but then we should target 2.6.2 with this PR